adam_core/coordinates/cartesian.py

```diff
@@ -82,126 +82,362 @@ def __init__(
         )
         return
 
+    @property
     def x(self):
         """
         X-coordinate
         """
         return self._values[:, 0]
 
+    @x.setter
+    def x(self, value):
+        self._values[:, 0] = value
+
+    @x.deleter
+    def x(self):
+        self._values[:, 0] = np.nan
+        self._values[:, 0].mask = True
+
+    @property
     def y(self):
         """
         Y-coordinate
         """
         return self._values[:, 1]
 
+    @y.setter
+    def y(self, value):
```

Comment:
When we set the value do we need to remove a possible mask?

Reply:
It really should remove the mask to make the behavior consistent with the deleter. I'll fix it!